### PR TITLE
Add TryFrom implementations

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Changelog
+
+## [v0.1.1](2023-24-02)
```
Comment (robin-nitrokey) [inline CHANGELOG.md]: missing `[] `

**Review by robin-nitrokey — APPROVED** [review]
LGTM!  If you do a new release anyway, can you please add `AsRef<[u8]>` and `AsMut<[u8]>` implementations?

**Comment by sosthene-nitrokey** [comment]
I also included an impl for `AsRef<[u8]>` and `Borrow<[u8]>` and their `mut` variants.

**Review by robin-nitrokey — APPROVED** [review]
Thanks!

**Pull request merged**